lock-api/src/test/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtilsTest.java

```diff
@@ -0,0 +1,154 @@
+            ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(PtBytes.toBytes("created")))));
+    private static final Random RAND = new Random();
+    private static ConjureMetadataConversionResult conjureMetadataConversionResult;
+
+    @BeforeClass
+    public static void setup() {
+        KeyListChecksum checksum =
+                IndexEncodingUtils.computeChecksum(ConjureLockRequestMetadataUtils.DEFAULT_CHECKSUM_TYPE, LOCK_LIST);
+        ConjureLockRequestMetadata conjureMetadata = ConjureLockRequestMetadata.builder()
+                .indexToChangeMetadata(CONJURE_LOCKS_WITH_METADATA)
+                .checksumTypeId(checksum.type().getId())
+                .checksumValue(Bytes.from(checksum.value()))
+                .build();
+        conjureMetadataConversionResult = ConjureMetadataConversionResult.builder()
+                .lockList(LOCK_LIST)
+                .conjureMetadata(conjureMetadata)
+                .build();
+    }
+
+    @Test
+    public void convertsToConjureCorrectly() {
+        assertThat(ConjureLockRequestMetadataUtils.toConjureIndexEncoded(LOCK_SET, LOCK_REQUEST_METADATA))
+                .isEqualTo(conjureMetadataConversionResult);
+    }
+
+    @Test
+    public void convertsFromConjureCorrectly() {
+        assertThat(ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(conjureMetadataConversionResult))
+                .isEqualTo(LOCK_REQUEST_METADATA);
+    }
+
+    @Test
+    public void convertingToAndFromConjureIsIdentityForRandomData() {
+        Set<LockDescriptor> lockDescriptors = Stream.generate(UUID::randomUUID)
+                .map(UUID::toString)
+                .map(StringLockDescriptor::of)
+                .limit(1000)
+                .collect(Collectors.toSet());
+        Map<LockDescriptor, ChangeMetadata> lockDescriptorToChangeMetadata = KeyedStream.of(lockDescriptors.stream())
+                .filter(_unused -> RAND.nextBoolean())
+                .map(_unused -> createRandomChangeMetadata())
+                .collectToMap();
+        LockRequestMetadata metadata = LockRequestMetadata.of(lockDescriptorToChangeMetadata);
+
+        assertThat(ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(
+                        ConjureLockRequestMetadataUtils.toConjureIndexEncoded(lockDescriptors, metadata)))
+                .isEqualTo(metadata);
+    }
+
+    @Test
+    public void changedLockOrderIsDetected() {
+        ConjureMetadataConversionResult conversionResult = ImmutableConjureMetadataConversionResult.copyOf(
+                        conjureMetadataConversionResult)
+                .withLockList(LOCK_2, LOCK_1, LOCK_3, LOCK_4);
+        assertThatThrownBy(() -> ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(conversionResult))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageStartingWith("Key list integrity check failed");
+    }
+
+    private static ChangeMetadata createRandomChangeMetadata() {
```

Comment:
Haven't fully looked at this yet, but we should use unit4j parameters, example: https://github.com/palantir/atlasdb/pull/6468/files#diff-6041f2c7bc69b08611eb2dce8e1d5f6d896ff70fbec0abe216648064fba7664aR33-R44

Reply:
What exactly would you paramaterise in this case? Which type of metadata is generated?

Reply:
With Junit4, we only have the option to parameterize the entire class. However, I'd like to keep some tests that mix different types of metadata in the same request (all tests do this right now).

Reply:
Yeah was more of thinking along the lines of, we should have the data be generated in a single place, rather than in method. Although, reading this now you only use it in a single place.

Reply:
Can we just ensure we throw all the random generation stuff into a single method, rather than splitting it across the test and method?

Reply:
Only other thing we need to make sure we do, is the comparison enough for us to debug what went wrong? IF not, it we get a random flake, it'll be really hard for us to try and determine what was broken :(

Reply:
We could (additionally) assert on the metadata map to get a nicer debug print from `AssertJ`. Also, we could seed our  `Random` instance before every test and print the seed in case the assertion fails.

Reply:
Actually, I will remove the `Random` instances from the class altogether and scope all the random generation within the method. This way, we also have direct control over the seed.